### ISSUE TITLE
Errors when strings contain reserved tokens

### ISSUE THREAD

**Issue by andrewgopher** [xyzqm/spark#18]
Leaving tokens in strings produces some peculiar errors; For example:
```
main() {
	vars:
		string: works_string;
	program:
		works_string = 'works#';
		print('{works_string}');
}
```
throws the error:
```
Invalid syntax: Expected RightBracket, got EndOfInput:
```
I believe this is because the interpreter thinks that this is a start of a comment, and since the "comment" stretches to the end of the file without matching the left bracket with a right bracket, it throws a syntax error. I'm not 100% sure, though

**Comment by xyzqm** [comment]
I'll look into it.

**Comment by xyzqm** [comment]
@andrew50git Ok, I've figured out the problem. If you look at the lexer, you'll see that it automatically skips over any hashtag, whether it's in a string literal or not. So, I'll create a token for the Hashtag and deal with it in the parser.

Good catch!

**Comment by xyzqm** [comment]
Actual after some tweaking, I've discovered it will be VERY HARD to parse a comment. So instead, of you want to use the # in a string, I will add a '\' escape key for you to use keywords.

**Comment by xyzqm** [comment]
#24 This pull request has updated the code to have an escape key.
